src/agent/feedback.ts

```diff
@@ -640,20 +658,30 @@ export class LogPageQueue extends BaseQueue<cm.ILogPageInfo> {
                                 logId = this._recordToLogIdMap[recordId];
                                 if (logId) {
                                     trace.write('uploading log page: ' + pagePath);
-                                    this._timelineApi.uploadLogFile(planId,
-                                        logId,
-                                        pagePath,
-                                        (err: any, statusCode: number, obj: any) => {
-                                            if (err) {
-                                                trace.write('error uploading log file: ' + err.message);
-                                            }
-
-                                            fs.unlink(pagePath, (err) => {
-                                                // we're going to continue here so we can get the next logs
-                                                // TODO: we should consider requeueing?
-                                                doneStep(null);
+                                    fs.stat(pagePath, (err, stats) => {
+                                        if (err) {
+                                            trace.write('Error reading log file: ' + err.message);
+                                            return;
+                                        }
+                                        var pageStream: NodeJS.ReadableStream = fs.createReadStream(pagePath);
+                                        this._taskApi.appendLog(
+                                            { "Content-Length": stats.size }, pageStream,
```

Comment:
see other comment.  ideally the generator would know that the logs controller requires a content-length header -- maybe could add an attribute.